topic-controller/src/main/java/io/strimzi/controller/topic/Controller.java

```diff
@@ -616,6 +616,63 @@ public void handle(Future<Void> fut) {
         inFlight.enqueue(topicName, resultHandler, futureHandler);
     }
 
+    void onTopicPartitionsChanged(TopicName topicName, Handler<AsyncResult<Void>> resultHandler) {
+        Handler<Future<Void>> futureHandler = new Reconciliation("onTopicPartitionsChanged") {
+            @Override
+            public void handle(Future<Void> fut) {
+
+                Handler<AsyncResult<TopicMetadata>> handler = new Handler<AsyncResult<TopicMetadata>>() {
+
+                    private final BackOff backOff = new BackOff();
+
+                    @Override
+                    public void handle(AsyncResult<TopicMetadata> metadataResult) {
+
+                        if (metadataResult.succeeded()) {
+
+                            // getting topic metadata from Kafka
+                            Topic kafkaTopic = TopicSerialization.fromTopicMetadata(metadataResult.result());
+
+                            // getting topic information from the private store
+                            topicStore.read(topicName, topicResult -> {
+
+                                // if partitions aren't changed on Kafka yet, we retry with exponential backoff
+                                if (topicResult.result().getNumPartitions() == kafkaTopic.getNumPartitions()) {
+
+                                    long delay;
+                                    try {
+                                        delay = backOff.delayMs();
+                                        logger.debug("Topic {} partitions changed, but metadata not updated in Kafka yet: Backing off for {}ms", topicName, delay);
+                                    } catch (MaxAttemptsExceededException e) {
+                                        logger.info("Topic {} partitions changed, but metadata not updated in Kafka after {}ms, giving up for now", topicName, backOff.totalDelayMs());
+                                        fut.fail(e);
```

Comment:
I wonder if we should `fut.fail()` here, given that we don't know if the znode changed due to partition increase or not. If we don't observe a partition count change within the backoff shouldn't we just assume it was a reassignment? I assume the completion of fut will complete the `resultHandler`, which will ultimately log and error and create a warning event.

Reply:
Agree with that. Updated with latest commit.

Reply:
Thanks